server/stream.go

```diff
@@ -201,6 +206,68 @@ func (a *Account) AddStreamWithStore(config *StreamConfig, fsConfig *FileStoreCo
 	return mset, nil
 }
 
+// Helper to determine the max msg size for this stream if file based.
+func (mset *Stream) maxMsgSize() uint64 {
+	maxMsgSize := mset.config.MaxMsgSize
+	if maxMsgSize <= 0 {
+		// Pull from server.
+		if mset.jsa != nil {
+			if acc := mset.jsa.acc(); acc != nil {
+				if s := acc.server(); s != nil {
+					opts := s.getOpts()
+					maxMsgSize = opts.MaxPayload
```

Comment:
Is the MaxPayload not possibly a limit set in the account JWT?

Reply:
Good point, should pull from internal client or account first. Thanks will update.